File: leetcode/python/0001_twosum_a.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nums = list(enumerate(nums))
        nums.sort(key=lambda e: e[1])
        
        a = 0
        b = len(nums) - 1
        
        while a < b:
            k = nums[a][1] + nums[b][1]
            if k > target:
                b = b - 1
            elif k < target:
                a = a + 1
            else:
                return [nums[a][0], nums[b][0]]

File: leetcode/scala/0001_twosum_a.scala

import scala.collection.mutable.HashMap

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val S: HashMap[Int, Int] = HashMap.empty[Int, Int]
        
        for (i <- nums.indices) {
            val x = target - nums(i)
            if (S.contains(x)) {
                val j = S.get(x).get
                return Array(j, i)
            } else {
                S.put(nums(i), i)
            }
        }
        return Array(0, 0)
    }
}

